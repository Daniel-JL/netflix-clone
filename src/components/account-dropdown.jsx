/* eslint-disable import/prefer-default-export */
import React, { useState } from 'react';
import {
  NavLink,
} from 'react-router-dom';
import styled from 'styled-components';
import { RectInfoButton } from './buttons';
import { AccountDropdownMenu } from './account-dropdown-menu';

export function AccountDropdown() {
  const [dropdownActive, setDropdownActive] = useState(false);

  return (
    <div>
      <nav>
        <RectInfoButton
          onMouseOver={() => setDropdownActive((dropdownActive) => !dropdownActive)}
          onMouseOut={() => setDropdownActive((dropdownActive) => !dropdownActive)}
        />
        {dropdownActive && <AccountDropdownMenu />}

      </nav>
    </div>
  );
}
