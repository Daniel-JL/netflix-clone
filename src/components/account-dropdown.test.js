import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { AccountDropdown } from './account-dropdown';

describe('account dropdown in navbar', () => {

  it('renders correctly', () => {
    const wrapper = renderer.create(<AccountDropdown />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });
  
  it('should show all available users', () => {

  });

  it('should have all options as clickable links', () => {

  });

  it('should activate on mouseover and deactivate on unmouseover', () => {

  });

  it('should activate/deactivate on click', () => {

  })
});
