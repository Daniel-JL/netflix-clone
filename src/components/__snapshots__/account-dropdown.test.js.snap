// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`account dropdown in navbar renders correctly 1`] = `
<div>
  <nav>
    <button
      className="sc-AxjAm sc-AxhCb fNFtln"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
  </nav>
</div>
`;
